runtime/codert_vm/cnathelp.cpp

```diff
@@ -885,13 +885,73 @@ old_fast_jitPutFlattenableStaticField(J9VMThread *currentThread)
 void* J9FASTCALL
 old_slow_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	SLOW_JIT_HELPER_PROLOGUE();
+	j9object_t arrayObject = (j9object_t)currentThread->floatTemp1;
+	U_32 index = *((U_32 *)currentThread->floatTemp2);
+	void *addr = NULL;
+	void *oldPC = currentThread->jitReturnAddress;
+	if (NULL == arrayObject) {
+		buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+		addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		j9object_t value = NULL;
+		U_32 arrayLength = J9INDEXABLEOBJECT_SIZE(_currentThread, arrayObject);
+		if (index >= arrayLength) {
+			currentThread->tempSlot = (UDATA)index;
```

Comment:
This is what is done on the ByteCodeInterpreter version of aaload  if the index is greater than the array length. Let me know if you think it is not needed.

Reply:
In the interpreter we have a special routine that will look at the tempSlot so that it can print a more helpful exception message. We dont have that here.